modules/Bio/EnsEMBL/DBSQL/AltAlleleGroupAdaptor.pm

```diff
@@ -231,20 +231,56 @@ sub fetch_by_dbID {
 sub fetch_by_gene_id {
     my ($self, $gene_id) = @_;
 
+    my $aag_list = $self->fetch_all_by_gene_id($gene_id);
+
+    # return first group from list
+    my $aag = @$aag_list[0];
+    # check that a result exists
+    if ($aag) {
+        my $group_id = $aag->dbID;
+        return $self->fetch_by_dbID($group_id);
+    }
+    return;
+}
+
+=head2 fetch_all_by_gene_id
+
+  Arg[1]      : Integer Gene ID of the member to query by
+  Description : Creates and returns an array of one or more AltAlleleGroups, 
+                which each contain the specified gene member                
+  Returntype  : Array of Bio::EnsEMBL::AltAlleleGroup objects
+
+=cut
+
+sub fetch_all_by_gene_id {
+    my ($self, $gene_id) = @_;
+
     my $gene_id_sql = q(
         SELECT alt_allele_group_id FROM alt_allele
         WHERE gene_id = ?
```

Comment:
The backwards-compatibility issue is a bit of a tricky one. This solution is probably as good as it gets, but I don't like that two calls to fetch_by_gene_id could return different alt-allele groups, depending on which one happens to be first in the list.  Consider adding an `ORDER BY alt_allele_group_id` here, so that the first alt-allele group in the returned list will be consistent. It may not be the one the user wants, but it will be much easier to track down the problem than if the error were intermittent!

Reply:
that's a very good point, and the order by should cover most of our use cases
in practice, as the PAR genes were added later, they will have a higher alt_allele_group_id, and we should not have that many cases in the first place (only happens for patch/haplotypes overlapping a PAR region)
we might get new cases in the future, although that will probably be a while

Reply:
Thanks very much for the ORDER BY suggestion. I've updated the SQL query to include this and therefore make the list of alt_allele_group_id values consistent.